operator-common/src/main/java/io/strimzi/operator/common/metrics/AbstractMetricsHolder.java

```diff
@@ -0,0 +1,230 @@
+     */
+    public AbstractMetricsHolder(String kind, Labels selectorLabels, MetricsProvider metricsProvider) {
+        this.kind = kind;
+        this.selectorLabels = selectorLabels;
+        this.metricsProvider = metricsProvider;
+    }
+
+    /**
+     * Metrics provider used for the metrics by this holder class
+     *
+     * @return  Metrics provider
+     */
+    public MetricsProvider metricsProvider()    {
+        return metricsProvider;
+    }
+
+    ////////////////////
+    // Methods for individual counters
+    ////////////////////
+
+    /**
+     * Counter metric for number of periodic reconciliations. It should be incremented only once per timer-trigger. It
+     * should not be incremented for every resource found by the periodical reconciliation.
+     *
+     * @param namespace     Namespace of the resources being reconciled
+     *
+     * @return  Metrics counter
+     */
+    public Counter periodicReconciliationsCounter(String namespace) {
+        return getCounter(namespace, kind, METRICS_PREFIX + "reconciliations.periodical", metricsProvider, selectorLabels, periodicReconciliationsCounterMap,
+                "Number of periodical reconciliations done by the operator");
+    }
+
+    /**
+     * Counter metric for number of reconciliations. Each reconciliation should increment it (i.e. it increments once
+     * per resource).
+     *
+     * @param namespace     Namespace of the resources being reconciled
+     *
+     * @return  Metrics counter
+     */
+    public Counter reconciliationsCounter(String namespace) {
+        return getCounter(namespace, kind, METRICS_PREFIX + "reconciliations", metricsProvider, selectorLabels, reconciliationsCounterMap,
+                "Number of reconciliations done by the operator for individual resources");
+    }
+
+    /**
+     * Counter metric for number of failed reconciliations.
+     *
+     * @param namespace     Namespace of the resources being reconciled
+     *
+     * @return  Metrics counter
+     */
+    public Counter failedReconciliationsCounter(String namespace) {
+        return getCounter(namespace, kind, METRICS_PREFIX + "reconciliations.failed", metricsProvider, selectorLabels, failedReconciliationsCounterMap,
+                "Number of reconciliations done by the operator for individual resources which failed");
+    }
+
+    /**
+     * Counter metric for number of successful reconciliations.
+     *
+     * @param namespace     Namespace of the resources being reconciled
+     *
+     * @return  Metrics counter
+     */
+    public Counter successfulReconciliationsCounter(String namespace) {
+        return getCounter(namespace, kind, METRICS_PREFIX + "reconciliations.successful", metricsProvider, selectorLabels, successfulReconciliationsCounterMap,
+                "Number of reconciliations done by the operator for individual resources which were successful");
+    }
+
+    /**
+     * Counter metric for number of resources.
```

Comment:
Can we clarify tha this is the number of consumed resources (i.e. CRs), rather than some counter over all resources (CRs plus the output/derived resources).